### PR TITLE
feat(markdown): Add markdown component

### PR DESCRIPTION
**Pull request by lordelogos**
**Pull Request Description:**

Following my conversation with @zenorocha regarding adding [md-to-react-email](https://www.npmjs.com/package/md-to-react-email) to core, this pull request introduces the following changes:

1. Added a Markdown to React-Email component to the `packages/markdown` directory.
2. Included documentation for the packages in the `docs/components` directory.

These changes aim to enhance the project by incorporating allowing users incorporate markdown directly into their React-Email template, improving the developer experience. The `packages/markdown` directory now contains the necessary files for the Markdown to React-Email component, while the `docs/components` directory provides comprehensive documentation for the packages.

These additions will contribute to better accessibility, maintainability, and understanding of the project.
I would also love to further conversation to better document this package as I am having difficulty setting up the docs locally on my machine

### CODE REVIEW

File: packages/markdown/src/markdown.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { render } from "@react-email/render";
+import { Markdown } from "./markdown";
+
+describe("ReactEmailMarkdown component renders correctly", () => {
+  it("renders the markdown in the correct format for browsers", () => {
+    const actualOutput = render(
```
Comment (bukinoshita) [inline packages/markdown/src/markdown.spec.tsx:6]: Can we add more tests cases? heading, text, link and list

Comment (lordelogos) [reply packages/markdown/src/markdown.spec.tsx:6]: On it!.

**Comment by bukinoshita** [comment]
Thanks a lot @lordelogos, this is huge 🖤

**Comment by lordelogos** [comment]
> Thanks a lot @lordelogos, this is huge 🖤

Thanks for reviewing @bukinoshita, I still have a couple of questions if you don't mind

- How will the package be published?
- Do I need to open another PR for documentation updates?
- How will I handle static assets for the docs?

**Pull request merged**